app/javascript/App/components/NewsfeedPage/NewsList.js

```diff
@@ -1,31 +1,84 @@
 import React, { Component, Fragment } from 'react'
+import _ from 'lodash'
 import NewsListItemAnimated from './NewsListItemAnimated'
 import LoadingIndicator from '../LoadingIndicator'
 
 class NewsList extends Component {
   state = { initialRender: true }
+
+  constructor(props) {
+    super(props)
+    this.mountOnScrollHandler = this.mountOnScrollHandler.bind(this)
+    this.unmountOnScrollHandler = this.unmountOnScrollHandler.bind(this)
+  }
+
   componentDidMount() {
     setTimeout(() => {
       this.setState({ initialRender: false })
     }, 6000)
+
+    this.mountOnScrollHandler()
+  }
+
+  componentDidUpdate() {
+    if (!window.isMobile && !window.isTablet) {
+      const $newsFeed = $('#news-feed')
+      const timer = setInterval(() => {
+        if ($newsFeed.height() < $newsFeed.find('> div').get(0).clientHeight) {
+          clearInterval(timer)
+        } else {
+          this.props.scrollBottomToReducer(true)
+        }
+      }, 1000)
+    }
+  }
+
+  componentWillUnmount() {
+    this.unmountOnScrollHandler()
+  }
+
+  mountOnScrollHandler() {
+    if (window.isMobile || window.isTablet) {
+      const throttled = _.throttle(this.props.onScrollNewsFeedMobile, 1000)
+      $(window).scroll(throttled)
+    } else {
+      const throttled = _.throttle(this.props.onScrollNewsFeedDesktop, 1000)
+      $('#news-feed').scroll(throttled)
+    }
   }
+
+  unmountOnScrollHandler() {
+    $(window).off('scroll', this.props.onScrollNewsFeedMobile)
+    $('#news-feed').off('scroll', this.props.onScrollNewsFeedDesktop)
+  }
+
   render() {
-    const { newsItems, isLoading } = this.props
+    const { isLoading } = this.props
+    const isMobile = window.isMobile
+
     const itemHeight = this.state.initialRender ? 'auto' : 0
     return (
       <Fragment>
         {isLoading('newsItems') && (
           <LoadingIndicator className="overlay bg-white-70" />
         )}
-        <div className="flex-auto overflow-y-auto relative">
-          {newsItems.map((newsItem) => (
-            <NewsListItemAnimated
-              key={newsItem.get('id')}
-              {...this.props}
-              newsItem={newsItem}
-              height={itemHeight}
-            />
-          ))}
+        <div id="news-feed"
+          className={'flex-auto relative overflow-y-' + (isMobile ? 'hidden' : 'auto')}>
```

Comment:
You can use Tachyons utilities with ns/m/l extension for applying a class to a screen size.
Eg. `overflow-y-hidden overflow-y-auto-m`
But in this case I think it should be set to auto regardless of the screen size